lib/server/publish_context.js

```diff
@@ -1,95 +1,81 @@
-PublishContext = function PublishContext(context, subscription) {
-  this.userId = context.userId;
-  this.unblock = function() {};
-  this._subscription = subscription;
-  this._context = context;
-  this._collectionData = {};
-  this._onStop = [];
-  this._stopped = false;
-
-  // connection object
-  this.connection = {
-    _id: Meteor.uuid(),
-    close: function() {},
-    onClose: function() {},
-    // fake value, will be supported later on
-    clientAddress: "127.0.0.1",
-    httpHeaders: context.headers
-  };
-
-  // we won't be supporting all the other fields of the Meteor's
-  // Subscription class since they are private variables
-};
-
-PublishContext.prototype._addCursor = function(cursor) {
+PublishContext = function PublishContext(context, handler, subscriptionId, params, name) {
   var self = this;
-  cursor.rewind();
-  var collectionName = 
-    (cursor._cursorDescription)? cursor._cursorDescription.collectionName: null || //for meteor-collections
-    (cursor._collection)? cursor._collection._name: null; //for smart-collections
```

Comment:
BTW, this is also something which was removed in the process. I do not know if this is problematic.